### ISSUE TITLE
Deleting a Monograph or Section should delete its children

### ISSUE THREAD

**Issue by conorom** [mlibrary/heliotrope#652]
This may simply be a byproduct of indexing issues, but it should be looked at.
When I run `Monograph.find('foobar123').destroy!`, more often than not I receive a `ActiveTriples::ParentStrategy::UnmutableParentError` error, as seen below.

Sometimes it works (which probably means some weird index stuff is happening?).
The monograph appears to be deleted, but some or all associated FileSets remain. Not sure about Sections.
Unfortunately, this is more likely to occur on staging/preview rather than dev, so I suspect the increased activity may bring about the (assumed) indexing issues.

```
nectar:20161206123459 conorom$ sudo -u heliotrope-staging RAILS_ENV="production" bundle exec rails console
/hydra-dev/heliotrope-staging/releases/20161206123459/config/routes.rb:1: warning: already initialized constant RESQUE_MOUNT_PATH
/hydra-dev/heliotrope-staging/releases/20161206123459/config/routes.rb:1: warning: previous definition of RESQUE_MOUNT_PATH was here
Loading production environment (Rails 4.2.7.1)
irb(main):001:0> Monograph.find('p8418n27h').destroy!
ActiveFedora: loading fedora config from /hydra-dev/heliotrope-staging/releases/20161206123459/config/fedora.yml
ActiveFedora: loading solr config from /hydra-dev/heliotrope-staging/releases/20161206123459/config/solr.yml
Attempted to init base path `heliotrope-staging`, but it already exists
ActiveTriples::ParentStrategy::UnmutableParentError: ActiveTriples::ParentStrategy::UnmutableParentError
	from /hydra-dev/heliotrope-staging/shared/vendor/bundle/ruby/2.3.0/gems/active-triples-0.11.0/lib/active_triples/persistence_strategies/parent_strategy.rb:95:in `parent='
	from /hydra-dev/heliotrope-staging/shared/vendor/bundle/ruby/2.3.0/gems/active-triples-0.11.0/lib/active_triples/rdf_source.rb:100:in `initialize'
	from /hydra-dev/heliotrope-staging/shared/vendor/bundle/ruby/2.3.0/gems/active-triples-0.11.0/lib/active_triples/rdf_source.rb:626:in `new'
	from /hydra-dev/heliotrope-staging/shared/vendor/bundle/ruby/2.3.0/gems/active-triples-0.11.0/lib/active_triples/rdf_source.rb:626:in `from_uri'
	from /hydra-dev/heliotrope-staging/shared/vendor/bundle/ruby/2.3.0/gems/active-triples-0.11.0/lib/active_triples/relation.rb:568:in `make_node'
	from /hydra-dev/heliotrope-staging/shared/vendor/bundle/ruby/2.3.0/gems/active-triples-0.11.0/lib/active_triples/relation.rb:454:in `convert_object'
	from /hydra-dev/heliotrope-staging/shared/vendor/bundle/ruby/2.3.0/gems/active-triples-0.11.0/lib/active_triples/relation.rb:314:in `block in each'
	from /hydra-dev/heliotrope-staging/shared/vendor/bundle/ruby/2.3.0/gems/rdf-2.1.0/lib/rdf/mixin/enumerable.rb:482:in `block in each_object'
	from /hydra-dev/heliotrope-staging/shared/vendor/bundle/ruby/2.3.0/gems/rdf-2.1.0/lib/rdf/mixin/queryable.rb:291:in `<<'
	from /hydra-dev/heliotrope-staging/shared/vendor/bundle/ruby/2.3.0/gems/rdf-2.1.0/lib/rdf/mixin/queryable.rb:291:in `block (2 levels) in enum_for'
	from /hydra-dev/heliotrope-staging/shared/vendor/bundle/ruby/2.3.0/gems/rdf-2.1.0/lib/rdf/repository.rb:412:in `block (4 levels) in query_pattern'
	from /hydra-dev/heliotrope-staging/shared/vendor/bundle/ruby/2.3.0/gems/hamster-3.0.0/lib/hamster/set.rb:166:in `block in each'
	from /hydra-dev/heliotrope-staging/shared/vendor/bundle/ruby/2.3.0/gems/hamster-3.0.0/lib/hamster/trie.rb:40:in `block in each'
	from /hydra-dev/heliotrope-staging/shared/vendor/bundle/ruby/2.3.0/gems/hamster-3.0.0/lib/hamster/trie.rb:40:in `each'
	from /hydra-dev/heliotrope-staging/shared/vendor/bundle/ruby/2.3.0/gems/hamster-3.0.0/lib/hamster/trie.rb:40:in `each'
	from /hydra-dev/heliotrope-staging/shared/vendor/bundle/ruby/2.3.0/gems/hamster-3.0.0/lib/hamster/set.rb:166:in `each'
... 11 levels...
	from /hydra-dev/heliotrope-staging/shared/vendor/bundle/ruby/2.3.0/gems/rdf-2.1.0/lib/rdf/mixin/enumerable.rb:478:in `each_object'
	from /hydra-dev/heliotrope-staging/shared/vendor/bundle/ruby/2.3.0/gems/active-triples-0.11.0/lib/active_triples/relation.rb:472:in `objects'
	from /hydra-dev/heliotrope-staging/shared/vendor/bundle/ruby/2.3.0/gems/active-triples-0.11.0/lib/active_triples/relation.rb:313:in `each'
	from /hydra-dev/heliotrope-staging/shared/vendor/bundle/ruby/2.3.0/gems/active-triples-0.11.0/lib/active_triples/relation.rb:39:in `to_a'
	from /hydra-dev/heliotrope-staging/shared/vendor/bundle/ruby/2.3.0/gems/active-triples-0.11.0/lib/active_triples/relation.rb:39:in `inspect'
	from /hydra-dev/heliotrope-staging/shared/vendor/bundle/ruby/2.3.0/gems/active-fedora-11.0.1/lib/active_fedora/attribute_methods.rb:184:in `attribute_for_inspect'
	from /hydra-dev/heliotrope-staging/shared/vendor/bundle/ruby/2.3.0/gems/active-fedora-11.0.1/lib/active_fedora/core.rb:91:in `block in inspect'
	from /hydra-dev/heliotrope-staging/shared/vendor/bundle/ruby/2.3.0/gems/active-fedora-11.0.1/lib/active_fedora/core.rb:90:in `collect'
	from /hydra-dev/heliotrope-staging/shared/vendor/bundle/ruby/2.3.0/gems/active-fedora-11.0.1/lib/active_fedora/core.rb:90:in `inspect'
	from /hydra-dev/heliotrope-staging/shared/vendor/bundle/ruby/2.3.0/gems/railties-4.2.7.1/lib/rails/commands/console.rb:110:in `start'
	from /hydra-dev/heliotrope-staging/shared/vendor/bundle/ruby/2.3.0/gems/railties-4.2.7.1/lib/rails/commands/console.rb:9:in `start'
	from /hydra-dev/heliotrope-staging/shared/vendor/bundle/ruby/2.3.0/gems/railties-4.2.7.1/lib/rails/commands/commands_tasks.rb:68:in `console'
	from /hydra-dev/heliotrope-staging/shared/vendor/bundle/ruby/2.3.0/gems/railties-4.2.7.1/lib/rails/commands/commands_tasks.rb:39:in `run_command!'
	from /hydra-dev/heliotrope-staging/shared/vendor/bundle/ruby/2.3.0/gems/railties-4.2.7.1/lib/rails/commands.rb:17:in `<top (required)>'
	from bin/rails:9:in `require'
```

**Comment by sethaj** [comment]
We've got a `destroy` method in the monograph model that should do this:

https://github.com/mlibrary/heliotrope/blob/master/app/models/monograph.rb#L50

Maybe do `Monograph.find('foobar123').destroy` (no `!`)? Or we could add a `destroy!` method to the model....

**Comment by conorom** [comment]
This seems to be a very consistent issue that occurs on nectar only.
I guess that points to a permissions issue that most likely would also occur on production, should we ever have to delete a monograph there.
It's not a big deal, but "orphaned" FileSets do seem to be left behind as a result.

**Comment by conorom** [comment]
This works just fine every time if you don't chain the command after `find`.
I'm going to guess you can't call `destroy`/`destroy!` in the console without holding a reference to the Work for the duration. Or something.

I can also confirm that the problem is the same in development. FileSets get orphaned if you chain the command, as seen up top...  only you don't get shown the `ActiveTriples::ParentStrategy::UnmutableParentError` in dev. Ever. Weird.

To sum up. Always do this:
```
m = Monograph.find('999999999')
m.destroy
```
Job done.
